platforms/core-configuration/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPluginCrossVersionSmokeTest.kt

```diff
@@ -222,6 +229,11 @@ class KotlinDslPluginCrossVersionSmokeTest : AbstractKotlinIntegrationTest() {
         }
     }
 
+    private
+    fun expectBuildScopeListenerDeprecation() {
```

Comment:
❓ Any chance of moving one of these utilities higher up and reuse wherever possible?

Reply:
We could, but that is not seems to be how we typically do it for tests. For tests, we seem to favor "code locality" over avoiding repetition across tests, at least for expected deprecations. I am usually bothered by repetition, but I can see the value of having the deprecation messages right there in the test case, or at least in the same file.